src/qt/walletmodel.cpp

```diff
@@ -449,6 +449,13 @@ void WalletModel::unsubscribeFromCoreSignals()
 // WalletModel::UnlockContext implementation
 WalletModel::UnlockContext WalletModel::requestUnlock()
 {
+    // Bugs in earlier versions may have resulted in wallets with private keys disabled to become "encrypted"
+    // (encryption keys are present, but not actually doing anything).
+    // To avoid issues with such wallets, check if the wallet has private keys disabled, and if so, return a context
+    // that indicates the wallet is not encrypted.
+    if (m_wallet->privateKeysDisabled()) {
```

Comment:
It's a pity that `getEncryptionStatus() == NoKeys` does not work here.

Is it worth to warn the user when they load a watch-only wallet with an encryption keys?

Reply:
> It's a pity that `getEncryptionStatus() == NoKeys` does not work here.

It could, but #631 states that `NoKeys` is not returned for encrypted watchonly so that other users of `getEncryptionStatus()` get this status correctly. Perhaps that isn't necessary, but I wanted to keep this change targeted for just this specific issue and changing `getEncryptionStatus()`'s behavior could have unanticipated side effects.

> Is it worth to warn the user when they load a watch-only wallet with an encryption keys?

https://github.com/bitcoin/bitcoin/pull/28724 proposes to just delete them for the user on loading.